feat: Date Range Selections

ApexTame opened the pull request:
- Added a Date Range Selections for clips.
- Note: idk how to fix an small issue i tried. but for Clips like in 2023 or further back, its Really slow. to Fetch The Clips. idk how to Really fix it. if you have an idea on how to fix it feel free to lmk. 
- Date Range Selection Ui is Based on Flowbite with some Tweaks.

ApexTame committed:
feat: Date Range Selections

- Added a Date Range Selections for clips.
- Note: idk how to fix an small issue i tried. but for Clips like in 2023 or further back, its Really slow. to load Clips. idk how to Really fix it. if you have an idea on how to fix it feel free to lmk.
- Date Range Selection Ui is Based on Flowbite with some Tweaks.

ApexTame committed:
fix: updated Version.

ApexTame committed:
fix: desktop app version.

- i forgot to update the desktop version.

Fedecas commented:
Nice contribution!

There is a bug when switching from date-range selection (with a range selected) to sorting by views: it shows no clips.

Unless we find a way to retrieve clips by date using the API itself, the current method for retrieving older clips will be slow; I don't think we can do much more for now.

ApexTame commented:
okay. ill fix the issue, but it'll will be later, im busy for a few hours, i'll try to get it done as soon as possable.
also do you like the green in the date range UI. cause idk if its me it kinda hard to tell the numbers color, i did it green so it matches the style of the project.

ApexTame committed:
fix: when sorting by views it shows date Selected Clips.

- i fixed it but for me it wasent show no clips it was showing the Date Selected clips. it should be fixed.

ApexTame commented:
okay i fixed. it i think it was showing me the date selected clips instead of views but it was not showing no clips at all. also we might need some preformace fixes. tbh i notics it lag when i was sorting by date range on xqc it lag bad. so we might need to do some preformaces updates. couple weeks ago to when i was using this tool. if you manually go to like a year ago clips without sorting by date range the Video player would be so laggy as well. if you want i can try and fix some issues.

Fedecas commented:
The color of the numbers in the UI could be darker when selected to provide more contrast and make them easier to read.

I don't notice any lag in the video player, but I do notice lag when sorting by specific date ranges on channels with many clips, such as xqc.
However, from what I've seen, we don't have any other option for implementing it, I believe.

ApexTame committed:
Ui: tweaked Colors when Selected.

- Tweaked Colors when Selected Dates to make it easier to Read.

ApexTame committed:
perf: made some Performance improvements

- I Made it so when Filtering By date Range it dose it in Batches, you can see it noticeably differents.

ApexTame commented:
okay. i somewhat fixed the lag when filtering. by Range. give it a test. also i made it so when selected dates it truns to a dark color with font-bold. to Performance i made it filter in batches. lmk what you think

ApexTame commented:
also question a while ago i coded a feat that is creator searched. witch is you can enter a kick username and it would load all the clips that username has created for the current channel your on. if you want i can add a new pr and add it, i would have to do some fixing with it but u can do that also for a new feat.

Fedecas commented:
Sorry about the delay in responding.

I see the changes you made, and they are good, but when I reach the bottom of the date-range selection, it clears all loaded clips and shows none.

ApexTame committed:
fix: Date Range Selection Clears all Loaded Clips

ApexTame commented:
alright i fixed it.

Fedecas commented:
Great, thanks for the contribution!
I'll give the project some love these days.
